src/ksm/scraper.go

```diff
@@ -154,7 +157,10 @@ func (s *Scraper) Run(i *integration.Integration) error {
 	}
 
 	if !populated {
-		return fmt.Errorf("KSM data was not populated after trying all endpoints")
+		if s.Filterer != nil {
+			return fmt.Errorf("%v, check if namespace filters are set properly", errNoData)
```

Comment:
%w?

Reply:
Yes, thanks for spotting it!, it works but it's not correct.

Reply:
Fixed in https://github.com/newrelic/nri-kubernetes/pull/493/commits/1caff07f4c9e8c1ea91ae567437481833b46c500